SmartDeviceLinkTests/DevAPISpecs/NSMutableDictionary+StoreSpec.m

```
@@ -0,0 +1,165 @@
+            }).to(raiseException());
+        });
+    });
+
+    describe(@"adding an array of custom objects", ^{
+        __block NSDictionary *testJson = @{};
+        __block TestObject *testObject = [[TestObject alloc] initWithDictionary:testJson];
+        __block NSArray<NSDictionary *> *testJsonArray = @[testJson];
+        __block NSArray<TestObject *> *testObjectArray = @[testObject];
+
+        beforeEach(^{
+            [testDictionary sdl_setObject:testJsonArray forName:testKey];
+        });
+
+        it(@"should return array", ^{
+            expect([testDictionary sdl_objectsForName:testKey ofClass:[TestObject class]]).to(equal(testObjectArray));
+        });
+        it(@"should raise exceprion", ^{
+            expectAction(^{
+                expect([testDictionary sdl_objectsForName:testKey ofClass:[NSNumber class]]).to(beNil());
+            }).to(raiseException());
+        });
+    });
+
+    describe(@"adding a SDLEnum", ^{
+        __block SDLEnum testObject = @"Object";
+
+        beforeEach(^{
+            [testDictionary sdl_setObject:testObject forName:testKey];
+        });
+
+        it(@"should return SDLEnum", ^{
+            expect([testDictionary sdl_enumForName:testKey error:nil]).to(equal(testObject));
+        });
+        it(@"should raise exception", ^{
+            expectAction(^{
+                expect([testDictionary sdl_enumsForName:testKey error:nil]).to(beNil());
+            }).to(raiseException());
+        });
+    });
+
+    describe(@"adding an array of SDLEnum", ^{
+        __block SDLEnum testObject = @"Object";
+        __block NSArray<SDLEnum> *testObjectArray = @[testObject];
+
+        beforeEach(^{
+            [testDictionary sdl_setObject:testObjectArray forName:testKey];
+        });
+
+        it(@"should return array of SDLEnum", ^{
+            expect([testDictionary sdl_enumsForName:testKey error:nil]).to(equal(testObjectArray));
+        });
+        it(@"should raise exceprion", ^{
```

@"should raise an exception when retrieved as an NSNumber"

---

fixed in 6f291ca